### PR TITLE
feat: add yaml as default for resource editor

### PR DESCRIPTION
**Pull request by sagarrchauhan**
### Description

Added feature to show YAML as default for resource editor. Data is still saved and retrieved as JSON.

PR references to #422

### CODE REVIEW

File: src/components/SchemaResourcePage/SchemaResource.tsx [hunk 1]
```diff
@@ -25,7 +27,11 @@ export function SchemaResource({
     if (!id) return;
     setLoadingState("loading");
     getResource(resourceInfo, id)?.then((res) => {
-      setResource(res.data[0]);
+      const specData = res.data[0]?.spec;
+      setResource({
+        ...res.data[0],
+        spec: specData && !isEmpty(specData) ? dump(specData) : ""
```
Comment (moshloop) [inline src/components/SchemaResourcePage/SchemaResource.tsx:33]: can we set sortKeys = true?

Comment (sagarrchauhan) [reply src/components/SchemaResourcePage/SchemaResource.tsx:33]: I did not understand the context. Do you want me to add a property called sortKeys to the resource object?

Comment (moshloop) [reply src/components/SchemaResourcePage/SchemaResource.tsx:33]: No - js-yaml supports an option called `sortKeys` when dumping

Comment (sagarrchauhan) [reply src/components/SchemaResourcePage/SchemaResource.tsx]: Added the option

**Pull request merged**